concordium-contracts-common/src/impls.rs

```diff
@@ -202,7 +202,7 @@ impl Deserial for ExchangeRate {
 }
 
 impl Serial for ModuleReference {
-    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> { self.as_ref().serial(out) }
+    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> { out.write_all(self) }
```

Comment:
How does this work? Does it use the `Deref` implementation implicitly? Can we elaborate.